tools/check-typo

```diff
@@ -127,9 +127,12 @@ IGNORE_DIRS="
     if $is_svn || $is_cmd_line; then :; else continue; fi
     svnrules=''
     if $is_svn; then
-      case `git check-attr binary "$f"` in
-        *'binary: set') continue;;
-      esac
+      # 4b825d is automatically in every repository (see cache.h)
+      # diff-tree is plumbing, so the output is stable and -\t- is the
+      # guaranteed output for a binary file only
+      EMPTY=4b825dc642cb6eb9a060e54bf8d69288fbee4904
+      git diff-tree --numstat $EMPTY HEAD -- "$f" | grep -q "^-[[:blank:]]-" \
+        && continue
```

Comment:
Have a nice time out!

Indeed, you could just comment above this piece of code:


```
Below a "well-known" plumbing command to detect whether
git regards a particular file as binary. This takes into account
.gitattributes, but also works if the file has been automatically
detected as binary by git.

EMPTY is the hash of an empty tree (try `git hash-object -t tree /dev/null`), as
a way to get `diff-tree` to print the whole tree state; its `--numstat` output then
prints a summary of each file, in a format such that two dashes in the first two
column indicate a file inferred to be a binary.
```